PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1732,8 +1722,44 @@ bool Notepad_plus::findInFiles()
 		patterns2Match.insert(patterns2Match.begin(), TEXT("*.*"));
 	}
 
-	vector<generic_string> fileNames;
-	getMatchedFileNames(dir2Search, patterns2Match, fileNames, isRecursive, isInHiddenDir);
+	if (not dir2Search [0])
+	{ // search workspace files
+		if (_findReplaceDlg.isProjectPanel_1() && _pProjectPanel_1)
+		{
+			_pProjectPanel_1->enumWorkSpaceFiles (NULL, patterns2Match, fileNames);
+		}
+		if (_findReplaceDlg.isProjectPanel_2() && _pProjectPanel_2)
+		{
+			_pProjectPanel_2->enumWorkSpaceFiles (NULL, patterns2Match, fileNames);
+		}
+		if (_findReplaceDlg.isProjectPanel_3() && _pProjectPanel_3)
+		{
+			_pProjectPanel_3->enumWorkSpaceFiles (NULL, patterns2Match, fileNames);
+		}
+		if (fileNames.size() == 0) return false;
+	}
+	else if (not ::PathFileExists(dir2Search))

```

Comment:
Please use modern C++ symbol.

Reply:
Can you elaborate, please? What symbol is not modern? I'm unable to see it right now.

Reply:
@Uhf7 I think it is the `not` operator alias. I remember I had similar comment.

Reply:
So '!' is preferred?

Reply:
Yes, `!` is preferred.  
Don and I went through this ourselves a while ago.  I like `not` because it is more readable (and Python uses it), but apparently `not` in C/(C++) is an antiquated notion, but supported.

Reply:
Maybe it is worth mentioning in  in [Coding style](https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/CONTRIBUTING.md#coding-style)?

Reply:
OK, I was confused, because some sites in the internet say

> In Modern C++, you can also use and, or and not as boolean operators which means &&, || and ! respectively which is identical to languages like Python. 

Source of this example is https://dev.to/delta456/modern-c-and-or-and-not-as-boolean-operators-2jgf, but I'm sure I have read it on other places too. 

So, the actually request here is not to use `not` modern C++?. Means `!` instead of `not`. But `not` is modern, `!` is old. Or I'm confused.

Reply:
When Don and I debated it, IIRC he proved to me that `not` was old-style, or at least well-enough to satisfy me then.  Since `!` is most widely used in the current code-base for Notepad++, it seems fine to stick with it; plus it is  what Don prefers.  If a wide group of people prefer `not` , @ArkadiuszMichalski 's suggestion to put something about using `!` / `&&` / `||` in the contributing guidelines is smart.

Reply:
Sorry, IIRC??? Don't know this shortcut. Please help.

Reply:
IIRC = If I Recall Correctly
I thought that one was very common.
(Here's where @ArkadiuszMichalski jumps on me because I got on him one time for using some lesser known abbreviation in a posting here)

Reply:
@sasumner , "IIRC", Thank you, I decrypt more and more of such shortcuts, but this one was not available on the Internet instantly.

So, the essence here is, that there is a different opinion of what is modern and what is not.

Since I don't care about writing `!` or `not`, I will change it to `!`. I hope, this was really the problem @donho meant. Thank to @mere-human, @sasumner and @ArkadiuszMichalski for the hints.

Reply:
@Uhf7 Yes, I confirm `!` is more modern operator than "not", I cannot find the link right now, but for the sake of coding style in Notepad++, please use "!" instead of "not".

Reply:
@donho Add to contributing guidelines?:

```
Please conform to existing coding style when contributing to the Notepad++ project.  Observe the code near your intended change, and attempt to maintain the style of that code.  As an example, the majority of the Notepad++ code uses `!` , `&&` and `||` operators instead of `not`, `and`, and `or`.  So if you submit code that uses `not`, etc., it won't be accepted.
```

Or similar...

Reply:
@sasumner 
Is it really necessary? It's just common sense to me.
OTOH, I will clean up all the `not` operators in the project - I didn't aware of it from the beginning (I did some "contribution" probably surly).

Reply:
@donho said "is it really necessary?"

Well, if it had just come up ONCE, I'd say no (reference our discussion some time ago concerning it).
But it came up earlier this week with @mere-human code.
And now it has resurfaced again with @Uhf7 code.
Needing an explanation each time.
As you like, draw your own conclusion as to if it is necessary.
But...there were other nice phrases in my suggestion...the parts about taking note of the code around you, and mimicking its style.
Again, always your choice.

Reply:
@sasumner 
OK, you're right.
Please do a PR for that.

Reply:
Done in https://github.com/notepad-plus-plus/notepad-plus-plus/commit/dbac42edd0f9c25bfdd8a90cf59ec4f701352f53
There were a bunch of "not" !!!

Reply:
@Uhf7 
Sorry for the conflict :(